File: laboratories/ppo/exercises/lab09/python/main.py
from src.Handler import Handler
from src.InputReader import InputReader
from src.exceptions.InputException import InputException


on = True
handler = Handler()
reader = InputReader()

while on:
    try:
        reader.read()
    except InputException as exception:
        if handler.handle(exception).should_break():
            on = False

File: laboratories/ppo/exercises/lab09/java/Main.java
import lab09.Handler;
import lab09.InputReader;

public class Main {
	public static void main(String[] args) {
		boolean on = true;
		Handler handler = new Handler();
		InputReader reader = new InputReader();

		while (on) {
			try {
				reader.read();
			} catch (Throwable exception) {
				if (handler.handle(exception).shouldBreak()) {
					on = false;
				}
			}
		}
	}
}
